BlueShift-iOS-SDK/BlueShiftAppDelegate.m

```diff
@@ -1004,10 +1004,30 @@ - (void)trackAlertDismiss {
     [[BlueShift sharedInstance] trackEventForEventName:kEventDismissAlert andParameters:nil canBatchThisEvent:YES];
 }
 
-/// SDK triggeres app_open event automatically on app launch and is controlled by the enableAppOpenTrackEvent  config flag
+/// SDK triggeres app_open event automatically when app is launched from killed state and is controlled by the enableAppOpenTrackEvent config flag
+/// @discussion The automatic app_open events can be throttled by setting time interval in secods to config.automaticAppOpenTimeInterval.
 - (void)trackAppOpenOnAppLaunch:(NSDictionary *)parameters {
-    if ([BlueShift sharedInstance].config.enableAppOpenTrackEvent) {
-        [self trackAppOpenWithParameters:parameters];
+    @try {
+        if ([BlueShift sharedInstance].config.enableAppOpenTrackEvent) {
+            if ([BlueShift sharedInstance].config.automaticAppOpenTimeInterval == 0) {
+                [self trackAppOpenWithParameters:parameters];
+            } else {
+                double lastAppOpenTimestamp = [[NSUserDefaults standardUserDefaults] doubleForKey:kBlueshiftLastAppOpenTimestamp];
```

Comment:
the value stored in user defaults can live between app sessions, right? it won't get deleted when the app is killed. correct?

Reply:
Yes, this value will persist until the app is not uninstalled.